File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 8/4/2026, 7:26:54 PM
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l = 0 , r = 0;
        int n = s.length();
        int maxi = 0;
        Set<Character>st = new HashSet<>();

        while(r < n){
            while(st.contains(s.charAt(r))){
                st.remove(s.charAt(l));
                l++;
            }
            maxi = Math.max(maxi , r - l + 1);
            st.add(s.charAt(r));
            r++;
        }
        return maxi;
    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.cpp
// Last updated: 8/4/2026, 7:26:50 PM
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.length();
        int l = 0;
        int r = 0;
        int maxLen = 0;
        map<char , int>mpp;

        while(r < n){
            mpp[s[r]]++;
            while(mpp[s[r]] > 1){
                mpp[s[l]]--;
                l++;
            }
            maxLen = max(r - l + 1 , maxLen);
            r++;
        }
        return maxLen;

    }
};